crates/rpc/rpc-convert/src/transaction.rs

```diff
@@ -137,10 +137,10 @@ pub trait RpcConvert: Send + Sync + Unpin + Clone + Debug + 'static {
 
     /// Creates a transaction environment for execution based on `request` with corresponding
     /// `cfg_env` and `block_env`.
-    fn tx_env<Spec>(
+    fn tx_env(
         &self,
         request: RpcTxReq<Self::Network>,
-        cfg_env: &CfgEnv<Spec>,
+        cfg_env_chain_id: u64,
```

Comment:
we cant really make that change because this can cause side-effects because then we create the cfgenv twice, so we need to keep this as is